### PR TITLE
Refactoring :

### PR DESCRIPTION
**Pull request by guaj**
- refactored some elements for performance
- Added errors when both passwords don't match

### CODE REVIEW

File: src/components/Authentication/SignUp.js [hunk 1]
```diff
@@ -115,19 +109,24 @@ export default function SignUp() {
...
+        ? (
         <Typography align="center" color="#DA3A16">
             {registrationError.message}
         </Typography>
-    ) :
-        (<React.Fragment/>);
+        )
+        : null), [registrationError]);
 
     const ProgramSelect = (
```
Comment (mnashat) [inline src/components/Authentication/SignUp.js:131]: Can this be created as a shared component? It exists in both SignUp.js and EditProfile.js. This is the only duplicate in this code, so should be a quick fix.

File: src/components/Profile/EditProfile.js [hunk 2]
```diff
@@ -32,63 +31,46 @@ export default function EditProfile() {
...
-        message: "Error, please try again later",
-        hasError: false
-    });
-    const [confirmPassword, setConfirmPassword] = React.useState({password: '', isEqualToPassword: false});
+    const newPassword = useRef('');
+    const userEmail = React.useMemo(() => JSON.parse(localStorage.getItem("email")), []);
+    const token =  React.useMemo(() => JSON.parse(localStorage.getItem("token")), []);
+
+    const [registrationError, setRegistrationError] = React.useState("");
+    const [confirmPassword, setConfirmPassword] = React.useState('');
```
Comment (mnashat) [inline src/components/Profile/EditProfile.js:39]: What is the use of confirmPassword here? It is being set but is not being used. Can you clarify its purpose please?

**Review by mnashat — CHANGES_REQUESTED** [review]
Looks good overall! Just take a quick look at the mentioned comments and let me know. Thanks!

File: src/components/Profile/EditProfile.js (hunk 2 above)
Comment (guaj) [reply src/components/Profile/EditProfile.js:39]: good catch, will remove. The only place i use the confirmPAssword is in the function that validates if its the same password. No need to have a state for this. can create the const locally

File: src/components/Authentication/SignUp.js (hunk 1 above)
Comment (guaj) [reply src/components/Authentication/SignUp.js:131]: yes,lost of page implement this error thing. We can create a custom one so its displays the errorMEssage for every page

Comment (mnashat) [reply src/components/Authentication/SignUp.js:131]: I'm referring to ProgramSelect const, not the error. The same const exists in 2 different files, so maybe we can put it in a separate file as a common resource to reduce duplicates.

**Comment by guaj** [comment]
Issues related : #51

**Review by mnashat — APPROVED** [review]
Great job!!

**Pull request merged**